### PR TITLE
[feat]: give AnyWorkflow a Workflow conformance

### PR DESCRIPTION
**Pull request by jamieQ**
### Motivation

the fact that `AnyWorkflow` is not itself a `Workflow` creates confusion, and some awkwardness in some of our testing facilities. we'd like to add such a conformance to address these issues, and prevent consumers from having to write such type-erasing wrappers themselves.

### Changes

- add a `Workflow` conformance to `AnyWorkflow`
- replace `RootWorkflow` with `AnyWorkflow` in `WorkflowHostingController`
- expose the wrapped workflow for use in observability code
- add/update tests

## Checklist

- [x] Unit Tests
~- [ ] UI Tests~ n/a
~- [ ] Snapshot Tests (iOS only)~ n/a
~- [ ] I have made corresponding changes to the documentation~ n/a

### CODE REVIEW

**Comment by kyleve** [comment]
cc @bencochran probably something you have opinions on

File: Workflow/Sources/AnyWorkflow.swift [hunk 1]
```diff
@@ -18,17 +18,24 @@
...
     }
 
     /// Initializes a new type-erased wrapper for the given workflow.
     public init<T: Workflow>(_ workflow: T) where T.Rendering == Rendering, T.Output == Output {
-        self.init(storage: Storage<T>(
-            workflow: workflow,
-            renderingTransform: { $0 },
-            outputTransform: { $0 }
-        ))
+        if let workflow = workflow as? AnyWorkflow<Rendering, Output> {
```
Comment (jamieQ) [inline Workflow/Sources/AnyWorkflow.swift:30]: is there a way to get the compiler to do this without having to perform the runtime check?

File: WorkflowUI/Sources/Hosting/WorkflowHostingController.swift [hunk 2]
```diff
@@ -150,25 +150,4 @@ public final class WorkflowHostingController<ScreenType, Output>: UIViewControll
     }
 }
 
-/// Wrapper around an AnyWorkflow that allows us to have a concrete
-/// WorkflowHost without WorkflowHostingController itself being generic
-/// around a Workflow.
-fileprivate struct RootWorkflow<Rendering, Output>: Workflow {
```
Comment (jamieQ) [inline WorkflowUI/Sources/Hosting/WorkflowHostingController.swift:156]: no longer necessary

**Review by mjohnson12 — APPROVED** [review]
Looks good to me

**Pull request merged**